disk/disk_test.go

```diff
@@ -134,21 +144,205 @@ func TestReadNonExistFile(t *testing.T) {
 		diskName  string
 		fileRoot  string
 		fileName  string
+		mkdir     bool
 		blockSize int64
 		offSet    int64
-		writeLen  int64
+		readLen   int64
 	}{
-		{"disk0", "nowhere_exist", "no", 4096, 0, 50},
+		// read file on non-exist path
+		{"disk0", "nowhere_exist", "no", false, 4096, 0, 50},
+		// read non-exist file on exist path
+		{"disk0", "nowhere_exist", "no", true, 4096, 0, 50},
 	}
+
 	for i, tt := range tests {
-		d := &Disk{Name: tt.diskName, Root: tt.fileRoot}
-		p := make([]byte, tt.writeLen)
-		for i := int64(0); i < tt.writeLen; i++ {
+		d := newTestDisk(tt.diskName, tt.fileRoot, tt.mkdir)
+		defer d.Remove("", true)
+
+		p := make([]byte, tt.readLen)
+		_, err := d.ReadAt(tt.fileName, p, tt.offSet)
+		if !os.IsNotExist(err) {
+			t.Errorf("%d: expect file not exist, got error = %v", i, err)
+		}
+	}
+}
+
+func TestRename(t *testing.T) {
+	tests := []struct {
+		diskName    string
+		fileNameOld string
+		fileNameNew string
+	}{
+		{"disk0", "old", "new"},
+		{"disk0", "old", "old"},
+	}
+
+	for i, tt := range tests {
+		writeLen := int64(50)
+		d := newTestDisk(tt.diskName, "rename", true)
+		defer d.Remove("", true)
```

Comment:
the reason is the defer will be done outside the scope and it usually leads to some bugs...